File: src/main/java/com/googongill/aditory/service/dto/category/LikeCategoryResult.java
package com.googongill.aditory.service.dto.category;

import com.googongill.aditory.domain.Category;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class LikeCategoryResult {
    private Long categoryId;
    private Integer likeCount;

    public static LikeCategoryResult of(Category category) {
        return LikeCategoryResult.builder()
                .categoryId(category.getId())
                .likeCount(category.getCategoryLikes().size())
                .build();
    }

}

File: src/main/java/com/googongill/aditory/service/dto/category/LikeCategoryResult.kt
package com.googongill.aditory.service.dto.category

import com.googongill.aditory.domain.Category

data class LikeCategoryResult(
    val categoryId: Long?,
    val likeCount: Int?
) {

    companion object {
        fun of(category: Category): LikeCategoryResult {
            return LikeCategoryResult(
                categoryId = category.id,
                likeCount = category.categoryLikes.size
            )
        }
    }

}
